### DISCUSSION TITLE
Pocketbase Middleware for Custom Auth guard

### DISCUSSION THREAD

**Discussion by Criztiandev** [pocketbase/pocketbase#4657] [Q&A]
Im currently creating a simple system using React and Pocketbase, the experiece is blazingly fast for my development process, but there is one of my problem is, I dont know how to create a middleware that check all protected route also even check of the admin token is still validated, Can you help me to do this ?  please provide also a code so I could , code review it, :<

**Comment by sidarth-23** [comment]
> Im currently creating a simple system using React and Pocketbase, the experiece is blazingly fast for my development process, but there is one of my problem is, I dont know how to create a middleware that check all protected route also even check of the admin token is still validated, Can you help me to do this ?  please provide also a code so I could , code review it, :<

Are you using vanilla react or react frameworks like nextjs ?

**Reply by Criztiandev** [reply]
im using react vite

**Comment by j0sh** [comment]
for most built-in collections, just use API rules

for custom protected routes in golang - use `apis.RequireRecordAuth`

```golang
import github.com/pocketbase/pocketbase/apis

app.OnBeforeServe().Add(func(e *core.ServeEvent) error {
  e.Router.GET("/my/path", func(c echo.Context) error {
    // do stuff here
    return nil
  }, apis.RequireRecordAuth("users_collection"))
  return nil
})
```

to request from JS
```
        const token = pb.authStore.token;
        fetch("https://pocketbase-host/my/path", {
          headers: { Authorization: token },
        });
```

see [documentation](https://pocketbase.io/docs/go-routing/#middlewares) - there are also equivalents for extending with JS

**Comment by gokulgseven** [comment]
https://pocketbase.io/docs/js-routing/#middlewares 

documentation to see how to create middlewares in the backend for custom api in javascript 

extract from documentation - see the doicumentation on the above link for more detail 
```
routerAdd("GET", "/hello", (c) => {
    return c.string(200, "Hello world!")
}, $apis.requireAdminAuth())
```

there are already middlewares created to check the authentication
